src/ape_http/providers.py

```diff
@@ -1,15 +1,103 @@
-from typing import Iterator
+import shutil
+from pathlib import Path
+from typing import Any, Dict, Iterator, List, Optional
+from urllib.parse import urlparse
 
+from eth_typing import HexStr
+from geth import DevGethProcess, LoggingMixin  # type: ignore
+from requests.exceptions import ConnectionError
+from requests.exceptions import ConnectionError as RequestsConnectionError
 from web3 import HTTPProvider, Web3
+from web3.exceptions import ContractLogicError as Web3ContractLogicError
 from web3.gas_strategies.rpc import rpc_gas_price_strategy
 from web3.middleware import geth_poa_middleware
+from web3.types import RPCEndpoint
 
-from ape.api import ProviderAPI, ReceiptAPI, TransactionAPI
+from ape.api import ReceiptAPI, TransactionAPI, TransactionStatusEnum
 from ape.api.config import ConfigItem
-from ape.exceptions import ProviderError, TransactionError, VirtualMachineError
-from ape.utils import get_tx_error_from_web3_value_error
+from ape.api.providers import TestProviderAPI
+from ape.exceptions import (
+    ContractLogicError,
+    OutOfGasError,
+    ProviderError,
+    RPCError,
+    TransactionError,
+)
+from ape.logging import logger
+from ape.utils import generate_dev_accounts, get_gas_estimation_revert_error_message
 
-DEFAULT_SETTINGS = {"uri": "http://localhost:8545"}
+DEFAULT_SETTINGS = {"uri": "http://localhost:8545", "chain_name": "devchain"}
+
+
+class EphemeralGeth(LoggingMixin, DevGethProcess):
```

Comment:
Note: Unitl https://github.com/ethereum/py-geth/pull/77 is merged and released, the `LoggingMixing` causes errors.